File: plugins/autorun/lua/autorun/test/exec.py
from rich import print
from time import sleep

ts = 0.5


def main():
    print(f'Hello from {__name__}')
    sleep(1)
    print(f'File: {__file__}')

    # ERROR OUT
    # print(1/0)


if __name__ == "__main__":
    main()

File: plugins/autorun/lua/autorun/test/exec.lua
local User = {
  new = function(self, o)
    o = o or {}
    setmetatable(o, self)
    self.__index = self

    return o
  end,

  __tostring = function(self) return 'Hello, my name is ' .. self.name .. ' and I am ' .. self.age .. ' years old' end,

  inc_age = function(self) self.age = self.age + 1 end,
}

function User:dec_age() self.age = self.age - 1 end

local user = User:new({ name = 'John', age = 20 })

print(user)
user:inc_age()
print(user)
user:dec_age()
print(user)

user:dec_age()
user:dec_age()
print(user)

-- lua falsy values: false and nil
-- 0 is truthy
if not true then
  print('0 is falsy')
end
